File: problems/easy/grading_students/__tests__/test_grading_students.c
#include <stdio.h>
#include <stdlib.h>
#include "../grading_students.c"

void test() {
    int grades[] = {73, 67, 38, 33};
    int n = 4;
    int* result = gradingStudents(grades, n);

    int expected[] = {75, 67, 40, 33};
    for (int i = 0; i < n; i++) {
        if (result[i] != expected[i]) {
            printf("❌ Test failed at index %d. Expected %d, got %d\n", i, expected[i], result[i]);
            exit(1);
        }
    }

    printf("✅ C test passed!\n");
    free(result);
}

int main() {
    test();
    return 0;
}

File: problems/easy/grading_students/__tests__/TestGradingStudents.java
package problems.easy.grading_students.__tests__;

import problems.easy.grading_students.GradingStudents;

public class TestGradingStudents {
    public static void main(String[] args) {
        int[] grades = {73, 67, 38, 33};
        int[] result = GradingStudents.gradingStudents(grades);
        int[] expected = {75, 67, 40, 33};

        for (int i = 0; i < grades.length; i++) {
            if (result[i] != expected[i]) {
                throw new AssertionError("❌ Test failed at index " + i + ". Expected " + expected[i] + ", got " + result[i]);
            }
        }

        System.out.println("✅ Java test passed!");
    }
}
